sklearn/tests/test_common.py

```diff
@@ -367,13 +367,11 @@ def _estimators_that_predict_in_fit():
 def test_pandas_column_name_consistency(estimator):
     _set_checking_parameters(estimator)
     with ignore_warnings(category=(FutureWarning)):
-        with pytest.warns(None) as record:
+        with warnings.catch_warnings():
+            warnings.simplefilter("error")
             check_dataframe_column_names_consistency(
                 estimator.__class__.__name__, estimator
             )
-        for warning in record:
-            assert "was fitted without feature names" not in str(warning.message)
```

Comment:
Here we don't want to ensure that there's no warning but only ensure that a specific warning with a specific message is not raised. We need to keep that.